packages/core/components/Puck/index.tsx

```diff
@@ -273,6 +273,52 @@ export function Puck({
     }
   }, []);
 
+  const toggleSidebars = (sidebar: "left" | "right") => {
+    const widerViewport = window.matchMedia("(min-width: 638px)").matches;
+    const sideBarVisible =
+      sidebar === "left" ? leftSideBarVisible : rightSideBarVisible;
+    const oppositeSideBar =
+      sidebar === "left" ? "rightSideBarVisible" : "leftSideBarVisible";
+
+    dispatch({
+      type: "setUi",
+      ui: {
+        [`${sidebar}SideBarVisible`]: !sideBarVisible,
+        ...(!widerViewport ? { [oppositeSideBar]: false } : {}),
+      },
+    });
+  };
```

Comment:
Perf: could use `useCallback` here to avoid recreating the function on each render